### PR TITLE
Update samples to .NET Core 2.1

### PR DESCRIPTION
**Pull request by richlander**
- Update to .NET Core 2.1
- Add instructions for ASP.NET Core, Docker and HTTPS

### CODE REVIEW

**Comment by MichaelSimons** [comment]
This PR fixes https://github.com/dotnet/dotnet-docker/issues/530

**Comment by Ullmie02** [comment]
jQuery / Bootstrap doesn't work with the ASP.NET Core sample:
`Failed to find a valid digest in the 'integrity' attribute for resource 'https://ajax.aspnetcdn.com/ajax/jquery/jquery-3.3.1.min.js' with computed SHA-256 integrity 'FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8='. The resource has been blocked.`
I think this is because the jQuery got updated from version 2.2.0 to 3.3.1 but the integrity value didn't change.

File: samples/aspnetapp/aspnetcore-docker-https.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+```console
+dotnet dev-certs https -ep %USERPROFILE%\.aspnet\https\aspnetapp.pfx -p crypticpassword
+dotnet dev-certs https --trust
+```
+
+Run Docker image with ASP.NET Core configured for HTTPS:
+
+```console
+docker run --rm -it -p 8000:80 -p 8001:443 -e ASPNETCORE_URLS="https://+;http://+" -e ASPNETCORE_HTTPS_PORT=8001 -e ASPNETCORE_ENVIRONMENT=Development -v %APPDATA%\microsoft\UserSecrets\:/root/.microsoft/usersecrets -v %USERPROFILE%\.aspnet\https:/root/.aspnet/https/ aspnetapp
```
[the remark this answers is not in the document]
Comment (MichaelSimons) [reply samples/aspnetapp/aspnetcore-docker-https.md]: That's a fair point.  I will call out that the "normal" port conflict error that I am accustom to is 

```
C:\Program Files\Docker\Docker\Resources\bin\docker.exe: Error response from daemon: driver failed programming external connectivity on endpoint aspnetcore_sample2 (a05cd83f01cf86748953db485dba95f09bc1da1b5892e6505e57c03a3e3fe018): Bind for 0.0.0.0:8000 failed: port is already allocated.
``` 

The error with the https samples was less obvious.

File: samples/aspnetapp/aspnetapp/Views/Shared/_Layout.cshtml [hunk 2]
```diff
@@ -51,7 +54,7 @@
         <script src="~/js/site.js" asp-append-version="true"></script>
     </environment>
     <environment exclude="Development">
-        <script src="https://ajax.aspnetcdn.com/ajax/jquery/jquery-2.2.0.min.js"
+        <script src="https://ajax.aspnetcdn.com/ajax/jquery/jquery-3.3.1.min.js"
```
Comment (natemcmaster) [inline samples/aspnetapp/aspnetapp/Views/Shared/_Layout.cshtml]: You'll need to update the integrity attribute as well

Comment (natemcmaster) [reply samples/aspnetapp/aspnetapp/Views/Shared/_Layout.cshtml]: Looks like a bug in aspnet templates. I've opened https://github.com/aspnet/templating/issues/520 to follow up on this

**Comment by richlander** [comment]
Thanks @natemcmaster. I was able to find/replace with your table and get to zero errors w/o issue. Works.

**Comment by richlander** [comment]
FYI to anyone watching. I launched ASP.NET Core in "Production" mode with: `dotnet run --no-launch-profile`

**Pull request merged**